engine/src/cmd/music.cpp

```diff
@@ -504,17 +547,17 @@ void Music::SkipRandSong( int whichlist, int layer )
 
 void Music::_SkipRandSong( int whichlist, int layer )
 {
-    if (!game_options.Music)
+    if (!game_options.Music){
         return;
-    if (this != NULL) {
-        if ( whichlist != NOLIST && whichlist >= 0 && whichlist < (int) playlist.size() ) {
-            lastlist = whichlist;
-            static bool random = XMLSupport::parse_bool( vs_config->getVariable( "audio", "shuffle_songs", "true" ) );
-            if ( playlist[whichlist].size() )
-                GotoSong( whichlist, random ? randInt( playlist[whichlist].size() ) : playlist[whichlist].counter++
-                          %playlist[whichlist].size(), true, layer );
-            else
-                fprintf( stderr, "Error no songs in playlist %d\n", whichlist );
+    }
+    if ( whichlist != NOLIST && whichlist >= 0 && whichlist < (int) playlist.size() ) {
+        lastlist = whichlist;
+        static bool random = XMLSupport::parse_bool( vs_config->getVariable( "audio", "shuffle_songs", "true" ) );
+        if ( playlist[whichlist].size() ){
+            GotoSong( whichlist, random ? randInt( playlist[whichlist].size() ) : playlist[whichlist].counter++
+                        %playlist[whichlist].size(), true, layer );
+        } else {
+            fprintf( stderr, "Error no songs in playlist %d\n", whichlist );
             return;
```

Comment:
The return can probably go away. There doesn't seem to be any loop here. Line 508/551 makes sense as it's short cutting the function; but this one on 518/561 doesn't make so much sense as if it's missed the function will naturally return any way, just like when going through the other side of the if-block.

The addition of the braces makes it more clear that it's an unnecessary statement that can be removed.

Reply:
Chopped out the return function